oak_attestation_verification/src/claims.rs

```diff
@@ -175,16 +177,103 @@ pub fn verify_validity_duration<T>(
 
 /// Checks that the given endorsement statement is a valid and has the correct
 /// claim type.
-pub fn validate_endorsement(
-    claim: &Statement<ClaimPredicate<EndorsementStatement>>,
-) -> Result<(), InvalidClaimData> {
+pub fn validate_endorsement(claim: &EndorsementStatement) -> Result<(), InvalidClaimData> {
     validate_claim(claim)?;
     if claim.predicate.claim_type != ENDORSEMENT_V2 {
         return Err(InvalidClaimData::ClaimType);
     }
     Ok(())
 }
 
+/// Assembles digests found in endorsement statement into a protocol buffer.
+pub fn get_digest(claim: &EndorsementStatement) -> anyhow::Result<HexDigest> {
+    if claim.subject.len() != 1 {
+        anyhow::bail!(
+            "expected a single endorsement subject, found {}",
+            claim.subject.len()
+        );
+    }
+
+    let mut digest = HexDigest {
+        psha2: "".to_owned(),
+        sha1: "".to_owned(),
+        sha2_256: "".to_owned(),
+        sha2_512: "".to_owned(),
+        sha3_512: "".to_owned(),
+        sha3_384: "".to_owned(),
+        sha3_256: "".to_owned(),
+        sha3_224: "".to_owned(),
+        sha2_384: "".to_owned(),
+    };
+    for (key, value) in &claim.subject[0].digest {
```

Comment:
More functionally this could be expressed as a fold:

```rust
claim.subject[0].digest.fold(HexDigest::default(), |acc, (key, value)| set_digest_field_from_map(acc, key, value))
```

(after defining `set_digest_field_from_map` auxiliary function somewhere, which would also be easily testable if desired)

Reply:
Thanks!!